sound/soc/sof/intel/hda-loader.c

```diff
@@ -384,6 +385,63 @@ static int hda_dsp_boot_imr(struct snd_sof_dev *sdev)
 	return ret;
 }
 
+int hda_dsp_ipc4_load_library(struct snd_sof_dev *sdev,
+			      struct snd_sof_module_library_info *lib_info)
+{
+	struct hdac_ext_stream *hext_stream;
+	struct firmware stripped_firmware;
+	struct snd_dma_buffer dmab;
+	struct sof_ipc4_msg msg = {};
+	int ret, ret1;
+
+	stripped_firmware.data = lib_info->fw->data + lib_info->fw_offset;
+	stripped_firmware.size = lib_info->fw->size - lib_info->fw_offset;
+
+	/* prepare DMA for code loader stream */
+	hext_stream = hda_cl_stream_prepare(sdev, HDA_CL_STREAM_FORMAT,
+					    stripped_firmware.size,
+					    &dmab, SNDRV_PCM_STREAM_PLAYBACK);
+	if (IS_ERR(hext_stream)) {
+		dev_err(sdev->dev, "DMA prepare failed\n");
+		return PTR_ERR(hext_stream);
+	}
+
+	memcpy(dmab.area, stripped_firmware.data, stripped_firmware.size);
+
+	msg.primary = hext_stream->hstream.stream_tag - 1;
+	msg.primary |= SOF_IPC4_MSG_TYPE_SET(SOF_IPC4_GLB_LOAD_LIBRARY);
+	msg.primary |= SOF_IPC4_MSG_DIR(SOF_IPC4_MSG_REQUEST);
+	msg.primary |= SOF_IPC4_MSG_TARGET(SOF_IPC4_FW_GEN_MSG);
+	msg.primary |= SOF_IPC4_GLB_LOAD_LIBRARY_LIB_ID(lib_info->id);
```

Comment:
is this just me or this code looks like copy paste with only LOAD_LIBRARY and lib_info->id added? Could we refactor the existing code to avoid duplication of this code loader?